Projects/UOContent/Network/Packets/IncomingPlayerPackets.cs

```diff
@@ -401,11 +404,21 @@ public static void DisplayGumpResponse(NetState state, SpanReader reader)
                 return;
             }
 
-            var switches = new int[switchCount];
+            scoped ReadOnlySpan<int> switches = default;
 
-            for (var i = 0; i < switches.Length; ++i)
+            if (switchCount > 0)
             {
-                switches[i] = reader.ReadInt32();
+                int byteCount = switchCount * 4;
+
+                switches = MemoryMarshal.Cast<byte, int>(reader.Buffer.Slice(reader.Position, byteCount));
+                reader.Seek(byteCount, SeekOrigin.Current);
+
+                if (BitConverter.IsLittleEndian)
+                {
+                    Span<int> reverseSwitches = stackalloc int[switchCount];
```

Comment:
Let's not use stackalloc (or at least have a conditional) when it is above 1024bytes, cause then someone could crash the server. I think the total stack limit is 1MB.